src/Form/DashproductsConfigurationDataConfiguration.php

```diff
@@ -0,0 +1,106 @@
+
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+/**
+ * Handles configuration data for dashproducts module configuration.
+ */
+final class DashproductsConfigurationDataConfiguration implements DataConfigurationInterface
+{
+    private const ALLOWED_VALUES = [5, 10, 20, 50];
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     */
+    public function __construct(ConfigurationInterface $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration(): array
+    {
+        return [
+            'show_last_order' => (int) $this->configuration->get('DASHPRODUCT_NBR_SHOW_LAST_ORDER'),
+            'show_best_seller' => (int) $this->configuration->get('DASHPRODUCT_NBR_SHOW_BEST_SELLER'),
+            'show_most_viewed' => (int) $this->configuration->get('DASHPRODUCT_NBR_SHOW_MOST_VIEWED'),
+            'show_top_searches' => (int) $this->configuration->get('DASHPRODUCT_NBR_SHOW_TOP_SEARCH'),
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateConfiguration(array $configuration): array
+    {
+        if ($this->validateConfiguration($configuration)) {
+            $this->configuration->set('DASHPRODUCT_NBR_SHOW_LAST_ORDER', (int) $configuration['show_last_order']);
+            $this->configuration->set('DASHPRODUCT_NBR_SHOW_BEST_SELLER', (int) $configuration['show_best_seller']);
+            $this->configuration->set('DASHPRODUCT_NBR_SHOW_MOST_VIEWED', (int) $configuration['show_most_viewed']);
+            $this->configuration->set('DASHPRODUCT_NBR_SHOW_TOP_SEARCH', (int) $configuration['show_top_searches']);
+        }
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validateConfiguration(array $configuration): bool
+    {
+        return (
+            // Last orders
+            isset($configuration['show_last_order'])
+            && is_numeric($configuration['show_last_order'])
+            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
+            // Best sellers
+            && isset($configuration['show_best_seller'])
+            && is_numeric($configuration['show_best_seller'])
+            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
+            // Most viewed
+            && isset($configuration['show_most_viewed'])
+            && is_numeric($configuration['show_most_viewed'])
+            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
+            // Top searches
+            && isset($configuration['show_top_searches'])
+            && is_numeric($configuration['show_top_searches'])
+            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
```

Comment:
```suggestion
            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
            // Best sellers
            && isset($configuration['show_best_seller'])
            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
            // Most viewed
            && isset($configuration['show_most_viewed'])
            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
            // Top searches
            && isset($configuration['show_top_searches'])
            && in_array((int) $configuration['show_last_order'], self::ALLOWED_VALUES)
```

You already check if data is in ALLOWED_VALUES, so don't care about if it's numeric or not